core/utils/metadata-utils/src/main/java/datawave/query/util/MetadataHelper.java

```diff
@@ -1692,6 +1692,73 @@ public Map<String,Long> getCountsForFieldsInDateRange(Set<String> fields, Set<St
         return fieldCounts;
     }
 
+    /**
+     * Get fields that have not been ingested within the date range (start and end dates are inclusive).
+     *
+     * @param fields
+     *            the fields
+     * @param datatypes
+     *            the datatypes
+     * @param beginDate
+     *            the start date
+     * @param endDate
+     *            the end date
+     * @param specialFields
+     *            special fields to exclude from search
+     * @return a set of missing fields from the given date range
+     */
+    public Set<String> getMissingFieldsInDateRange(Set<String> fields, Set<String> datatypes, String beginDate, String endDate, Set<String> specialFields)
+                    throws TableNotFoundException {
+        SortedSet<String> sortedDatatypes = new TreeSet<>(datatypes);
+        Set<String> foundFields = new HashSet<>();
+        fields = Sets.difference(fields, specialFields);
+        Set<Range> ranges = createExactFieldCountRanges(fields);
+        StringBuilder dataTypeRegex = new StringBuilder();
+
+        if (ranges.isEmpty()) {
+            return Collections.emptySet();
+        }
+
+        int index = 0;
+        for (String dataType : sortedDatatypes) {
+            if (index < sortedDatatypes.size() - 1) {
+                dataTypeRegex.append(dataType).append("\u0000.*").append("|");
+                index++;
+            } else {
+                dataTypeRegex.append(dataType).append("\u0000.*");
+            }
+        }
+
+        AccumuloClient client = accumuloClient;
+        if (client instanceof WrappedAccumuloClient) {
+            client = ((WrappedAccumuloClient) client).getReal();
+        }
+
+        try (BatchScanner bs = ScannerHelper.createBatchScanner(client, getMetadataTableName(), getAuths(), fields.size())) {
+            IteratorSetting regexIter = new IteratorSetting(50, "regexFilter", RegExFilter.class);
+            if (!dataTypeRegex.toString().isEmpty()) {
+                regexIter.addOption(RegExFilter.COLQ_REGEX, dataTypeRegex.toString());
+            }
+            bs.setRanges(ranges);
+            bs.addScanIterator(regexIter);
+
+            for (Entry<Key,Value> entry : bs) {
+                try {
+                    DateFrequencyMap map = new DateFrequencyMap(entry.getValue().get());
```

Comment:
It is not guaranteed that all 'f' rows will have an aggregated value. It's possible for there to be rows that have not yet been aggregated, and thus have long values. See `MetadataHelper.getCardinalityForField(String fieldName, String datatype, Date begin, Date end)` for an example of how to determine if we're looking at an aggregated entry vs. a single-value one.